src/parser/internal/definition/modifier.rs

```diff
@@ -57,3 +61,10 @@ pub fn collect(state: &mut State) -> ParseResult<ModifierGroupDefinition> {
         modifiers,
     })
 }
+
+pub fn collect_some(state: &mut State) -> ParseResult<Option<ModifierGroupDefinition>> {
+    match collect(state) {
+        Ok(modifiers) if !modifiers.modifiers.is_empty() => Ok(Some(modifiers)),
+        _ => Ok(None),
+    }
+}
```

Comment:
it's okay to have an empty list of modifiers on the structure, we don't need an option.